modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/AutoFill.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.models.servicetemplate;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.eclipse.xpanse.modules.models.service.deploy.enums.DeployResourceKind;
+
+/**
+ * Define autofill properties for deployment variables.
+ */
+@Data
+public class AutoFill {
+
+    @NotNull
+    @Schema(description = "Deployed resource type")
+    private DeployResourceKind deployResourceKind;
+
+    @NotNull
+    @Schema(description = "Not whether the variable is allowed to be created")
```

Comment:
schema can be - defines if the required cloud resource can be newly created or should the existing resources must only be used.